src/mower_logic/src/mower_logic/behaviors/UndockingBehavior.cpp

```diff
@@ -63,7 +75,14 @@ Behavior *UndockingBehavior::execute() {
   exePathGoal.tolerance_from_action = true;
   exePathGoal.controller = "DockingFTCPlanner";
 
-  auto result = mbfClientExePath->sendGoalAndWait(exePathGoal);
+  auto result = waitForResultOrAborted(mbfClientExePath, exePathGoal);
+  mbfClientExePath->sendGoal(exePathGoal);
```

Comment:
That's already in the helper.

Reply:
Yeah, I think this might have been the cause of docking failing for me.

Reply:
This was in undocking - but yeah, maybe there were side-effects on the docking as well.